src/support/modify.c

```diff
@@ -316,18 +315,17 @@ __wt_modify_apply_item(
   WT_SESSION_IMPL *session, const char *value_format, WT_ITEM *value, const void *modify)
 {
     WT_MODIFY mod;
-    size_t datasz, destsz, tmp;
+    size_t datasz, destsz, napplied, nentries;
     size_t item_offset;
-    const size_t *p;
-    int napplied, nentries;
+    const uint8_t *p;
     bool overlap, sformat;
 
     /*
      * Get the number of modify entries and set a second pointer to reference the replacement data.
      */
     p = modify;
-    memcpy(&tmp, p++, sizeof(size_t));
-    nentries = (int)tmp;
+    memcpy(&nentries, p, sizeof(size_t));
+    p += sizeof(size_t);
```

Comment:
Nitpick: I think all of the `sizeof(size_t)` can be replaced with a `sizeof(variable_name)`, e.g. this one could just be `sizeof(nentries)`. This way we only have to know the type in one spot, at declaration.

Reply:
I have done that, I wasn't sure if we should have applied this to the `WT_MODIFY_FOREACH_BEGIN` and `WT_MODIFY_FOREACH_REVERSE` macros here or not.